command.py

```diff
@@ -1,24 +1,70 @@
+#!/usr/bin/env python3
+
 import argparse
-import os
+import logging
+import pathlib
 import pkgutil
+import os
 
-if __name__ == '__main__':
-
-    COMMANDS_DIRECTORY = 'commands'
+import coloredlogs
+import verboselogs
 
-    parser = argparse.ArgumentParser(description='Managing %s commands' % os.environ.get("PROJECT_NAME"))
-    subparsers = parser.add_subparsers(help='sub commands')
 
-    subcommands = []
-    for _, command_name, _ in pkgutil.iter_modules([COMMANDS_DIRECTORY]):
-        path = "%s.%s" % (COMMANDS_DIRECTORY, command_name)
-        commands_import = __import__(path)
-        subcommands.append(getattr(commands_import, command_name))
+def main(project_name: str):
+    # Discover all the commands defined in derived image
+    parser = argparse.ArgumentParser(description=f"Managing {project_name} commands")
+    parser.add_argument(
+        "--log-level",
+        choices=sorted(logging._nameToLevel, key=lambda name: logging._nameToLevel[name]),
+        default="INFO",
+    )
 
-    for subcommand in subcommands:
-        subcommand.parse_args(subparsers)
+    # Import our commands and ask them to populate the subparsers with their argument configurations
+    gather_commands(
+        subparsers=parser.add_subparsers(help="Sub-commands"),
+        commands_directory=pathlib.Path(os.environ.get("ALADDIN_COMMANDS_PATH", "/code/commands")),
+    )
 
+    # Actually parse the arguments
     args = parser.parse_args()
 
+    # Use the provided log level and upgrade our logging capabilities
+    coloredlogs.install(
+        level=logging._nameToLevel[args.log_level],
+        fmt="%(levelname)-8s %(name)s(%(lineno)d) %(message)s",
+        level_styles=dict(
+            spam=dict(color="white", faint=True),
+            debug=dict(color="black", bold=True),
+            verbose=dict(color="blue"),
+            info=dict(color="white"),
+            notice=dict(color="magenta"),
+            warning=dict(color="yellow"),
+            success=dict(color="green", bold=True),
+            error=dict(color="red"),
+            critical=dict(color="red", bold=True),
+        ),
+        field_styles=dict(
+            asctime=dict(color="green"),
+            hostname=dict(color="magenta"),
+            levelname=dict(color="white"),
+            name=dict(color="white", bold=True),
+            programname=dict(color="cyan"),
+            username=dict(color="yellow"),
+        ),
+    )
+
+    # Invoke the requested command
     args.func(args)
 
+
+def gather_commands(subparsers: argparse.Action, commands_directory: pathlib.Path):
+    """Search for available commands and gather their arguments."""
```

Comment:
Also, question... how does ALADDIN_COMMANDS_PATH get into sys.path?  I feel a little dense... I'm not sure I see how the modules we gather here are importable with __import__ unless commands_directory is in the default location....

Reply:
Thanks for calling this out. As mentioned above, I'm abandoning the `__import__()` call in favor of finder.find_module().load_module(). This lets us pull in modules from outside the system path.